internal/pkg/cli/svc_deploy.go

```diff
@@ -378,11 +381,20 @@ func (o *deploySvcOpts) manifest() (interface{}, error) {
 	return mft, nil
 }
 
+func envUsesLegacySvcDiscovery(cf envCFDescriber, app, env string) (bool, error) {
+	return cf.EnvironmentUsesLegacySvcDiscovery(app, env)
+}
+
 func (o *deploySvcOpts) runtimeConfig(addonsURL string) (*stack.RuntimeConfig, error) {
+	svcDiscovery, err := envUsesLegacySvcDiscovery(o.envCFN, o.appName, o.envName)
+	if err != nil {
+		return nil, err
+	}
 	if !o.buildRequired {
 		return &stack.RuntimeConfig{
-			AddonsTemplateURL: addonsURL,
-			AdditionalTags:    tags.Merge(o.targetApp.Tags, o.resourceTags),
+			AddonsTemplateURL:      addonsURL,
+			AdditionalTags:         tags.Merge(o.targetApp.Tags, o.resourceTags),
+			LegacyServiceDiscovery: svcDiscovery,
```

Comment:
I wonder if this fits under `RuntimeConfig`. The original idea behind this struct was to capture additional configuration passed by the client when calling `deploy`: 1) the url for `addons/` template 2) if they use the `deploy --resource-tags` flag.

Maybe we can push down the `LegacyServiceDiscovery` one level below into the `stack` pkg? so that the clients have to do less work